beetsplug/convert.py

```diff
@@ -298,21 +307,46 @@ def convert_item(self, dest_dir, keep_new, path_formats, fmt,
                                    util.displayable_path(original))
                     util.move(item.path, original)
 
+            linked = False
+
             if should_transcode(item, fmt):
                 try:
                     self.encode(command, original, converted, pretend)
                 except subprocess.CalledProcessError:
                     continue
             else:
-                if pretend:
-                    self._log.info(u'cp {0} {1}',
-                                   util.displayable_path(original),
-                                   util.displayable_path(converted))
+                if hardlink:
+                    if pretend:
+                        self._log.info(u'ln {0} {1}',
+                                       util.displayable_path(original),
+                                       util.displayable_path(converted))
+                    else:
+                        # No transcoding necessary.
+                        self._log.info(u'Hardlinking {0}',
+                                       util.displayable_path(item.path))
+                        util.hardlink(original, converted)
+                        linked = True
+                elif link:
+                    if pretend:
+                        self._log.info(u'ln -s {0} {1}',
+                                       util.displayable_path(original),
+                                       util.displayable_path(converted))
+                    else:
+                        # No transcoding necessary.
+                        self._log.info(u'Linking {0}',
+                                       util.displayable_path(item.path))
+                        util.link(original, converted)
+                        linked = True
```

Comment:
Maybe we can hoist this assignment out and just write `linked = link or hardlink`?

Reply:
Also, is there any way to factor out some of the logic here so there's less duplicated code? For example, perhaps we could assign to a string to switch between `'ln'` and `'ln -s'` but only keeping one copy of the `_log.info` line.

Reply:
`link` and `hardlink` are set globally for the entire command. This way, copying album art still occurs for files that actually were converted, and is only skipped on those that are linked.